model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/solve_nonhydro.py

```diff
@@ -617,6 +543,10 @@ def __init__(
         self._stencils_42_44_45_45b = nhsolve_stencils.stencils_42_44_45_45b.with_backend(
             self._backend
         )
+        if self._config.divdamp_type == 32:
+            self.starting_vertical_index_for_3d_divdamp = np.min(
+                np.where(self._metric_state_nonhydro.scaling_factor_for_3d_divdamp.asnumpy() > 0.0)
+            )
```

Comment:
Use the backend dependent import form data_allocation.py instead of numpy. It should be somthing like:
```suggestion
xp = data_alloc.import_array_ns(self.backend)
self.starting_vertical_index_for_3d_divdamp = xp.min(
              xp.where(self._metric_state_nonhydro.scaling_factor_for_3d_divdamp.ndarray > 0.0)
            )
```